### ISSUE TITLE
installing from git error 

### ISSUE THREAD

**Issue by raptoria** [rrag/react-stockcharts#21]
Sorry..still having trouble getting this to work -.-**

Since I have a proxy issue with one of the dependencies, the npm install react-stockcharts times out.
Now, I am trying to do git clone, and then run npm install and then npm build inside the package. Unfortunately, I am getting this error now:

ERROR in ./src/index.js
Module parse failed: /home/clacey/react-flask/node_modules/react-stockcharts/src/index.js Line 4: Unexpected token
You may need an appropriate loader to handle this file type.
| 
| // common components
| import ChartCanvas from "./lib/ChartCanvas";
| import Chart from "./lib/Chart";
| import DataSeries from "./lib/DataSeries";

I have looked at the webpack and it has babel specified as a loader for .js and .jsx files, so I'm a little confused ..

**Comment by rrag** [comment]
Is this your set up?
```
react-flask
  |-package.json
  |-...
  |-...
  |-node_modules
    |-react-stockcharts (cloned from git) - This does not have a node_modules folder inside it
    |-react
    |-object-assign
    |-save-svg-as-png
    |-d3
    |-...
    |-...
```

If so post the loaders section of your webpack config

**Comment by raptoria** [comment]
inside react stockcharts package.json, I changed the dependencies to:
    "save-svg-as-png": "https://github.com/exupero/saveSvgAsPng.git#gh-pages"
    "prismjs": "https://github.com/PrismJS/prism.git#gh-pages",

your webpack file loaders section:

	module: {
		loaders: [
			{ test: /\.json$/, loader: "json" },
			{ test: /\.(js|jsx)$/, loaders: ["babel"], exclude: /node_modules/ },
		]
	},


my webpack loaders section:

	var loaders = {
		"jsx": options.hotComponents ? ["react-hot-loader", "babel-loader?stage=0"] : "babel-loader?stage=0",
		"js": {
			loader: "babel-loader?stage=0",
			include: path.join(__dirname, "src")
		},


	module: {
			preLoaders: [
			      {
			        //Eslint loader
			        test: /\.(js|jsx)$/,
			        loader: 'eslint-loader',
			        include: [path.resolve(__dirname, "src/")],
			        exclude: [path.resolve(__dirname, 'node_modules')]
			      },
			    ],
			loaders: loadersByExtension(loaders).concat(loadersByExtension(stylesheetLoaders)).concat(additionalLoaders)
		},

**Comment by rrag** [comment]
Your .js loader includes only your src folder. Try including node_modules/react-stockchsrts in addition to your src.

Also why do you not have the includes in the .jsx loader?

**Comment by raptoria** [comment]
ah..thanks for the tips. I got rid of the error now, but I have a new one.. 

Uncaught Error: Invariant Violation: ChartCanvas.render(): A valid ReactComponent must be returned. You may have returned undefined, an array or some other invalid object.

my two classes:


        import React, {PropTypes} from 'react';
        import CandleStick from "./CandleStick"

        let ReactD3  = require('react-stockcharts');

        export default class PnlChart extends React.Component {

           static propTypes = {
             closable: PropTypes.bool.isRequired,
             title:PropTypes.string.isRequired,
           }
           
           static defaultProps = { 
              closable: true,
              title:"Pnl Chart",
            }

            state = {
              data: []
            }


            componentDidMount(){
              let parseDate = d3.time.format("%Y-%m-%d").parse;
              d3.tsv("//rrag.github.io/react-stockcharts/data/MSFT.tsv", (err, data) => {
                /* change MSFT.tsv to MSFT_full.tsv above to see how this works with lots of data points */
                data.forEach((d, i) => {
                  d.date = new Date(parseDate(d.date).getTime());
                  d.open = +d.open;
                  d.high = +d.high;
                  d.low = +d.low;
                  d.close = +d.close;
                  d.volume = +d.volume;
                  // console.log(d);
                });
                  console.log("Got the data");
                  this.setState({ data });

              });
           }

          render() {
                  let data = this.state.data;
                  console.log("Pnl Chart is rendering");
            
            return (
                  <div>
                      { ( data.length > 0 ) ? <CandleStick data={data} type="svg" /> : " "}
                  </div>
            );
           }

        }





        import React, {PropTypes} from 'react';

        let ReactD3  = require('react-stockcharts');

        let { ChartCanvas, Chart, DataSeries, OverlaySeries, EventCapture } = ReactD3;

        let { CandlestickSeries, HistogramSeries, LineSeries, AreaSeries, StochasticSeries } = ReactD3.series;
        let { MouseCoordinates, CurrentCoordinate } = ReactD3.coordinates;
        let { EdgeContainer, EdgeIndicator } = ReactD3.coordinates;

        let { TooltipContainer, OHLCTooltip, MovingAverageTooltip, StochasticTooltip } = ReactD3.tooltip;
        let { StockscaleTransformer } = ReactD3.transforms;

        let { XAxis, YAxis } = ReactD3.axes;
        let { MACD, EMA, SMA, FullStochasticOscillator } = ReactD3.indicator;
        let { ChartWidthMixin, TypeChooser } = ReactD3.helper;


        export default class CandleStick extends React.Component {

           static propTypes = {
             data: React.PropTypes.array.isRequired,
             type: React.PropTypes.oneOf(["svg", "hybrid"]).isRequired
           }
           
           static defaultProps = { 
              type: "svg",
              data: []
            }

          render() {
            //var width = this.props.width || this.state !== null && this.state.width;
            let width = 500;
            //if (!width) return <div />;
            var { data, type } = this.props;
            var dateFormat = d3.time.format("%Y-%m-%d");
            return (
              <ChartCanvas width={width} height={750}
                margin={{left: 70, right: 70, top:20, bottom: 30}} initialDisplay={200} 
                dataTransform={[ { transform: StockscaleTransformer } ]}
                data={data} type={type}>
                <Chart id={1} yMousePointerDisplayLocation="right" height={325}
                    yMousePointerDisplayFormat={(y) => y.toFixed(2)} padding={{ top: 10, right: 0, bottom: 20, left: 0 }}>
                  <YAxis axisAt="right" orient="right" ticks={5} />
                  <XAxis axisAt="bottom" orient="bottom" showTicks={false} outerTickSize={0} />
                  <DataSeries id={0} yAccessor={CandlestickSeries.yAccessor} >
                    <CandlestickSeries />
                  </DataSeries>
                  <DataSeries id={1} indicator={EMA} options={{ period: 26 }} >
                    <LineSeries/>
                  </DataSeries>
                  <DataSeries id={2} indicator={EMA} options={{ period: 12 }} >
                    <LineSeries/>
                  </DataSeries>
                </Chart>
                <CurrentCoordinate forChart={1} forDataSeries={1} />
                <CurrentCoordinate forChart={1} forDataSeries={2} />
                <Chart id={2} yMousePointerDisplayLocation="left" yMousePointerDisplayFormat={d3.format(".4s")}
                    height={100} origin={(w, h) => [0, h - 475]} >
                  <YAxis axisAt="left" orient="left" ticks={5} tickFormat={d3.format("s")}/>
                  <DataSeries id={0} yAccessor={(d) => d.volume} >
                    <HistogramSeries fill={(d) => d.close > d.open ? "#6BA583" : "red"} />
                  </DataSeries>
                  <DataSeries id={1} indicator={SMA} options={{ period: 10, pluck:"volume" }} stroke="steelblue" fill="steelblue">
                    <AreaSeries opacity={0.5} />
                  </DataSeries>
                </Chart>
                <CurrentCoordinate forChart={2} forDataSeries={0} />
                <CurrentCoordinate forChart={2} forDataSeries={1} />
                <EdgeContainer>
                  <EdgeIndicator itemType="last" orient="right"
                    edgeAt="right" forChart={1} forDataSeries={1} />
                  <EdgeIndicator itemType="last" orient="right"
                    edgeAt="right" forChart={1} forDataSeries={2} />
                  <EdgeIndicator itemType="first" orient="left"
                    edgeAt="left" forChart={1} forDataSeries={1} />
                  <EdgeIndicator itemType="first" orient="left"
                    edgeAt="left" forChart={1} forDataSeries={2} />
                </EdgeContainer>
                <Chart id={3} yMousePointerDisplayLocation="right" yMousePointerDisplayFormat={(y) => y.toFixed(2)}
                    height={125} origin={(w, h) => [0, h - 375]} padding={{ top: 10, right: 0, bottom: 10, left: 0 }} >
                  <XAxis axisAt="bottom" orient="bottom" showTicks={false} outerTickSize={0} />
                  <YAxis axisAt="right" orient="right" ticks={2}/>
                  <DataSeries id={1} indicator={FullStochasticOscillator} options={{ period: 14, K: 1, D: 3 }} >
                    <StochasticSeries />
                  </DataSeries>
                </Chart>
                <Chart id={4} yMousePointerDisplayLocation="right" yMousePointerDisplayFormat={(y) => y.toFixed(2)}
                    height={125} origin={(w, h) => [0, h - 250]} padding={{ top: 10, right: 0, bottom: 10, left: 0 }} >
                  <XAxis axisAt="bottom" orient="bottom" showTicks={false} outerTickSize={0} />
                  <YAxis axisAt="right" orient="right" ticks={2}/>
                  <DataSeries id={1} indicator={FullStochasticOscillator} options={{ period: 14, K: 3, D: 3 }} >
                    <StochasticSeries />
                  </DataSeries>
                </Chart>
                <Chart id={5} yMousePointerDisplayLocation="right" yMousePointerDisplayFormat={(y) => y.toFixed(2)}
                    height={125} origin={(w, h) => [0, h - 125]} padding={{ top: 10, right: 0, bottom: 10, left: 0 }} >
                  <XAxis axisAt="bottom" orient="bottom"/>
                  <YAxis axisAt="right" orient="right" ticks={2}/>
                  <DataSeries id={1} indicator={FullStochasticOscillator} options={{ period: 14, K: 3, D: 3 }} >
                    <StochasticSeries />
                  </DataSeries>
                </Chart>
                <MouseCoordinates xDisplayFormat={dateFormat} type="crosshair" />
                <EventCapture mouseMove={true} zoom={true} pan={true} mainChart={1} defaultFocus={false} />
                <TooltipContainer>
                  <OHLCTooltip forChart={1} origin={[-40, -10]}/>
                  <MovingAverageTooltip forChart={1} onClick={(e) => console.log(e)} origin={[-38, 5]} />
                  <StochasticTooltip forChart={3} origin={[-38, 15]}>Fast STO</StochasticTooltip>
                  <StochasticTooltip forChart={4} origin={[-38, 15]}>Slow STO</StochasticTooltip>
                  <StochasticTooltip forChart={5} origin={[-38, 15]}>Full STO</StochasticTooltip>
                </TooltipContainer>
              </ChartCanvas>
              );

           }

        }


any ideas?? >_<

**Comment by rrag** [comment]
It is hard to tell. you should try with a simple chart first, and may be add some debug statements inside ChartCanvas.jsx to figure out what is being returned.

or may be since you are using source instead of npm published I might have pushed some untested updates up.

**Comment by raptoria** [comment]
I removed all the code from the render() function to only return an empty div, and still got the same error. I then moved one of my working components into the package and exported it from there - same error. A very strange and inexplicable bug..

Is there any chance you can use https instead of git:// in your package.json? This is what a few other NPM modules do to get around the proxy issue. maybe if there's some way I can install the package normally, this will all go away. 

"save-svg-as-png": "https://github.com/exupero/saveSvgAsPng.git#gh-pages"
"prismjs": "https://github.com/PrismJS/prism.git#gh-pages",

**Comment by rrag** [comment]
I don't see how that would help. prismjs is a dev dependency and that is removed before I publish to npm.

I am sure there is some other problem, I cant help troubleshoot it over this issue. may be you can come on gitter.im and we can chat there.

**Comment by rrag** [comment]
added https://gitter.im/rrag/react-stockcharts

**Comment by rrag** [comment]
published `0.2.6` to npm using 

```
"save-svg-as-png": "1.0.1"
```
